testing/testrunner/eventstrace.go

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: Elastic-2.0
+
+/*
+ * Copyright 2022 Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under
+ * one or more contributor license agreements. Licensed under the Elastic
+ * License 2.0; you may not use this file except in compliance with the Elastic
+ * License 2.0.
+ */
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os/exec"
+	"time"
+)
+
+type EventsTraceInstance struct {
+	Cmd        *exec.Cmd
+	Stdout     io.ReadCloser
+	Stderr     io.ReadCloser
+	StdoutChan chan string
+	StderrChan chan string
+}
+
+const eventsTraceBinPath = "/EventsTrace"
+
+func (et *EventsTraceInstance) Start() {
+	if err := et.Cmd.Start(); err != nil {
+		fmt.Println("failed to start EventsTrace: ", err)
+		TestFail()
+	}
+
+	et.StdoutChan = make(chan string)
+	go func() {
+		scanner := bufio.NewScanner(et.Stdout)
+		for scanner.Scan() {
+			et.StdoutChan <- scanner.Text()
+		}
+	}()
+
+	et.StderrChan = make(chan string, 100)
+	go func() {
+		scanner := bufio.NewScanner(et.Stderr)
+		for scanner.Scan() {
+			et.StderrChan <- scanner.Text()
+		}
+	}()
+
+	// This timeout is long on purpose, when running without KVM on a system
+	// under heavy load, it may actually take tens of seconds for EventsTrace
+	// to start up
+	select {
+	case <-et.StdoutChan:
+		break
+	case <-time.After(60 * time.Second):
+		et.DumpStderr()
+		TestFail("timed out waiting for EventsTrace to get ready, dumped stderr above")
+	}
+}
+
+func (et *EventsTraceInstance) DumpStderr() {
+	fmt.Println("===== EventsTrace Stderr =====")
+	for {
+		select {
+		case line, ok := <-et.StderrChan:
+			if !ok {
+				return
+			}
+			fmt.Println(line)
+		case <-time.After(1 * time.Second):
+			return
+		}
+	}
+}
+
+func (et *EventsTraceInstance) GetNextEventJson(types ...string) string {
+	var line string
+loop:
+	for {
+		select {
+		case line = <-et.StdoutChan:
+			eventType := getJsonEventType(line)
+
+			for _, a := range types {
+				if a == eventType {
+					break loop
```

Comment:
Personal opinion: I think that is not a good place to use goto lables in Go.
```suggestion
					return line
```

Reply:
Agreed.

Reply:
How would you re-write this? `return line` on line 99 necessitates a kinda wonky `return ""` at the end of the function that will never be reached, but needs to be there to satisfy the compiler? IMO this is slightly better.